staging_vespalib/src/tests/clock/clock_benchmark.cpp

```diff
@@ -0,0 +1,171 @@
+// Copyright 2017 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#include <vespa/vespalib/util/clock.h>
+#include <cassert>
+#include <vector>
+#include <atomic>
+#include <cstring>
+
+using vespalib::Clock;
+using fastos::TimeStamp;
+
+struct UpdateClock {
+    virtual ~UpdateClock() {}
+    virtual void update() = 0;
+};
+
+struct NSValue : public UpdateClock {
+    void update() override { _value = std::chrono::steady_clock::now().time_since_epoch().count(); }
+    int64_t _value;
+};
+
+struct NSVolatile : public UpdateClock {
+    void update() override { _value = std::chrono::steady_clock::now().time_since_epoch().count(); }
+    volatile int64_t _value;
+};
+struct NSAtomic : public UpdateClock {
+    void update() override { _value.store(std::chrono::steady_clock::now().time_since_epoch().count()); }
```

Comment:
As @havardpe mentioned, this will incur an artificially high performance impact since a plain `atomic::store` incurs a full memory store/load fence in all cases. Should probably make memory order for the store operation parameterized to match that of the load in the test body (i.e. `seq_cst` -> `seq_cst`, `release` -> `acquire` etc)

Reply:
I agree, but that is not a dimension that is interesting to test here. So I just wanted too keep it simple.
I am only interested in read performance, not write. Igf you are updating with frequnce of 1Mhz or higher it would matter, but not when using 100hz-100khz.